pkg/envoy/lds/response.go

```diff
@@ -63,21 +62,8 @@ func NewResponse(meshCatalog catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_d
 
 	// --- INBOUND -------------------
 	inboundListener := newInboundListener()
-	if meshFilterChain, err := getInboundInMeshFilterChain(proxyServiceName, cfg); err != nil {
-		log.Error().Err(err).Msgf("Error making in-mesh filter chain for proxy %s", proxy.GetCommonName())
-	} else {
-		if !cfg.IsPermissiveTrafficPolicyMode() {
-			// Apply RBAC policies on the inbound filters based on configured policies
-			rbacFilter, err := lb.buildRBACFilter()
-			if err != nil {
-				log.Error().Err(err).Msgf("Error applying RBAC filter for service %s", proxyServiceName)
-				return nil, err
-			}
-			// RBAC filter should be the very first filter in the filter chain
-			meshFilterChain.Filters = append([]*xds_listener.Filter{rbacFilter}, meshFilterChain.Filters...)
-		}
-		inboundListener.FilterChains = append(inboundListener.FilterChains, meshFilterChain)
-	}
+	inboundMeshFilterChains := lb.getInboundMeshFilterChains(proxyServiceName)
+	inboundListener.FilterChains = append(inboundListener.FilterChains, inboundMeshFilterChains...)
```

Comment:
Is there a reason for doing an append here instead of setting inboundListener.FilterChains to inboundMeshFilterChains? I just worry about duplicates.

Reply:
this just avoids relying relying on the order of building different filer chains (prometheus, ingress, in-mesh etc.).
I don't think there will be any duplicates here, and we have to use append regardless elsewhere since we continue to add filter chains to the same listener.